### PR TITLE
fix(icon): HttpClientModule removed from icon service

### LINKED ISSUE #4285: Angular httpinterceptor(jwt token header) not working after importing IgxTreeGridModule in lazy loaded module
### Description  
Angular httpinterceptor(jwt token header) not working after importing IgxTreeGridModule in lazy loaded module

- Ignite UI CLI version: no
- Framework: Angular
- Project type: igx-ts

### Steps to reproduce  

1. create lazy loaded module 
2. import  IgxTreeGridModule in the module 

### Result  
in the app module httpinterceptor is working 
image
in the lazy loaded module the unterceptor stop 

image

### Expected result  
if we remove IgxTreeGridModule from import list in the lazy loaded it works 
image

### code for this issue
https://github.com/badre429/igx-jwt-error

[or live on stackblitz](https://stackblitz.com/github/badre429/igx-jwt-error)

### PR DESCRIPTION
**Pull request by sboykova**
Closes #4285 

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/icon/icon.service.ts [hunk 1]
```diff
@@ -128,18 +132,40 @@ export class IgxIconService {
...
-            this.cacheSvgIcon(iconName, value, fontSet);
-        }, (error) => {
-            throw new Error(`Could not fetch SVG from url: ${url}; error: ${error.message}`);
-        }, () => {
-            subscription.unsubscribe();
+        this._httpRequest = new XMLHttpRequest();
+        const args = { iconName, url, fontSet };
+        this.zone.runOutsideAngular(() => {
+            this._loadListener = this.onLoad.bind(this, args);
+            this._httpRequest.addEventListener('load', this._loadListener);
```
Comment (bkulov) [inline projects/igniteui-angular/src/lib/icon/icon.service.ts]: Please use `onLoad` instead of attaching event listener.

Comment (sboykova) [reply projects/igniteui-angular/src/lib/icon/icon.service.ts]: Event listener removed

File: projects/igniteui-angular/src/lib/icon/icon.service.spec.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
-import { TestBed, async } from '@angular/core/testing';
+import { TestBed } from '@angular/core/testing';
 import { IgxIconService } from './icon.service';
-import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/icon/icon.service.spec.ts]: You should be able to remove the client testing module now.

File: projects/igniteui-angular/src/lib/icon/icon.service.spec.ts [hunk 3]
```diff
@@ -54,27 +54,22 @@ describe('Icon Service', () => {
...
         const iconService = TestBed.get(IgxIconService) as IgxIconService;
-        const httpMock = TestBed.get(HttpTestingController);
         const document = TestBed.get(DOCUMENT);
 
         const iconName = 'test';
         const fontSet = 'svg-icons';
         const iconKey = fontSet + '_' + iconName;
 
+        spyOn(XMLHttpRequest.prototype, 'open').and.callThrough(); // Jasmine 2.x
+        spyOn(XMLHttpRequest.prototype, 'send');
```
Comment (bkulov) [inline projects/igniteui-angular/src/lib/icon/icon.service.spec.ts]: Please spy on onLoad too.

**Pull request merged**